scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+        kill -9 $NETCAT_KILL
+        if [ "$?" != "0" ]; then
+            wsrep_log_debug "-> Cannot kill the existing NetCat process PID $NETCAT_KILL"
+        fi
+    fi
+
+    # OPEN NETCAT to receive messages from DONOR (like IST)
+    wsrep_log_info "-> Opening NETCAT: nc -l $JOINER_CLONE_HOST $JOINER_CLONE_PORT"
+    nc -l $JOINER_CLONE_HOST $JOINER_CLONE_PORT > $WSREP_SST_OPT_DATA/XST_FILE.txt &
+    NC_PID=$!
+    wsrep_log_info "-> NETCAT PID $NC_PID"
+    if [ $NC_PID == "" ];then
+        wsrep_log_error "-> Cannot open Netcat at given port $JOINER_CLONE_HOST $JOINER_CLONE_PORT check if the port is already taken"
+        exit 1
+    fi
+
+    # Report clone credentials/address to the caller
+    wsrep_log_debug "-> ready passing string |$CLONE_USER:CLONE_PSWD@$JOINER_CLONE_HOST:$JOINER_CLONE_PORT|"
+    echo "ready $CLONE_USER:$CLONE_PSWD@$JOINER_CLONE_HOST:$JOINER_CLONE_PORT"
+
+    # WAIT for Donor message
+    wsrep_log_debug "-> wait $JOINER_TIMEOUT_WAIT_XST"
+
+    until grep -q ".*<EOF>$" "$WSREP_SST_OPT_DATA/XST_FILE.txt" &> /dev/null
+    do
+         if [ "$JOINER_TIMEOUT_WAIT_XST" == "0" ]; then
+            wsrep_log_error "************ FATAL ERROR ************"
+            wsrep_log_error "TIMEOUT Waiting for the DONOR MESSAGE"
+            wsrep_log_error "Donor message was either incomplete or not sent"
+            donor_message=`cat $WSREP_SST_OPT_DATA/XST_FILE.txt`
+            wsrep_log_error "donor message received: $donor_message"
+            wsrep_log_error "*************************************"
+            exit 1 ;
+         fi
+         sleep 1
+
+        ((JOINER_TIMEOUT_WAIT_XST-=1))
+    done
+
+    wsrep_log_debug "-> WAIT DIR DONOR MESSAGE DONE"
+
+    if ! grep -q "SST@" "$WSREP_SST_OPT_DATA/XST_FILE.txt"; then
+        wsrep_log_info "DONOR SAY IST"
+        wsrep_log_debug "-> RECOVER POSITION TO SEND OUT DONOR IST"
+         RP_PURGED=`cat $WSREP_SST_OPT_DATA/XST_FILE.txt`
+         RP_PURGED=${RP_PURGED%"<EOF>"}
+         wsrep_log_debug "-> POSITION: $RP_PURGED"
+         rm -f  $WSREP_SST_OPT_DATA/XST_FILE.txt || :
+
+         echo $RP_PURGED
+         exit 0
+    else
+         RP_PURGED=`cat $WSREP_SST_OPT_DATA/XST_FILE.txt`
+         wsrep_log_info "DONOR SAY SST ($RP_PURGED)"
+         readonly CLEANUP_CLONE_PLUGIN=${RP_PURGED%|SST*}
+         RP_PURGED_EMERGENCY=${RP_PURGED#*SST@}
+         RP_PURGED_EMERGENCY=${RP_PURGED_EMERGENCY%"<EOF>"}
+         wsrep_log_debug "-> CLONE Plugin to remove? (yes/no) = $CLEANUP_CLONE_PLUGIN"
+         wsrep_log_debug "-> recovered position from DONOR: $RP_PURGED_EMERGENCY"
+    fi
+
+    ##################################################################################################
+    # If we need to SST in any case we must remove the data, so let us do it here and be sure we work on a clean directory
+    wsrep_log_info "Cleaning Data directory $WSREP_SST_OPT_DATA"
+    rm -fr $WSREP_SST_OPT_DATA/*
```

Comment:
SC2115: Use "${var:?}" to ensure this never expands to /*

Reply:
Yeah, this is good one. Nothing better than rm -rf /*
:)